msteams-platform/messaging-extensions/how-to/search-commands/respond-to-search.md

```diff
@@ -91,9 +91,55 @@ Your service must respond with the results matching the user query. The response
+The configuration response is the data returned by the server or application to configure and enable the message extension within the messaging platform. A `config` response is used when a user configures the message extension for the first time. 
+It triggers a prompt for the user to set up the message extension and provide any necessary configuration.
+
+If a message extension uses a configuration page, the [`onQuery`](/dotnet/api/microsoft.teams.ai.messageextensions-1.onquery) handler should first verify for any stored configuration data. If the message extension isn't configured, a `config` response must be returned, which includes a link to your configuration.
+
+The [`onQuery`](/dotnet/api/microsoft.teams.ai.messageextensions-1.onquery) handler also manages the response from the configuration page. The `onQuerySettingsUrl` handler provides the URL for the configuration page. After closing the configuration page, the `onSettingsUpdate` method is called to accept and save the returned state. The [`onQuery`](/dotnet/api/microsoft.teams.ai.messageextensions-1.onquery) handler then retrieves the updated settings and uses them to update the behavior of the message extension.
+
+The following code handles a user request for the configuration page of a message extension. It fetches the user's existing configuration settings, escapes them, and builds a response. This response includes the configuration page's URL with the escaped settings added as a query parameter.
+
+```csharp
+
+protected override async Task<MessagingExtensionResponse> OnTeamsMessagingExtensionConfigurationQuerySettingUrlAsync(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionQuery query, CancellationToken cancellationToken)
+{
+    // The user has requested the Messaging Extension Configuration page.
+    var escapedSettings = string.Empty;
+    var userConfigSettings = await _userConfigProperty.GetAsync(turnContext, () => string.Empty);
+    if (!string.IsNullOrEmpty(userConfigSettings))
+    {
+        escapedSettings = Uri.EscapeDataString(userConfigSettings);
+    }
+    return new MessagingExtensionResponse
+    {
+        ComposeExtension = new MessagingExtensionResult
+            {
+                Type = "config",
+                SuggestedActions = new MessagingExtensionSuggestedAction
+                    {
+                        Actions = new List<CardAction>
+                            {
+                                new CardAction
+                                {
+                                    Type = ActionTypes.OpenUrl,
+                                    Value = $"{_siteUrl}/searchSettings.html?settings={escapedSettings}",
+                                },
+                            },
+                    },
+            },
+    };
+}
+
+```
+
+The following image shows the `config` command workflow:
```

Comment:
Diagrams like this should go *before* code. (Lay down the foundational concepts first before going into detailed code.)

Reply:
Updated